src/Lykke.AlgoStore.Api/RealTimeStreaming/DataStreamers/WebSockets/Middleware/CandlesWebSocketsMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Lykke.AlgoStore.Api.RealTimeStreaming.DataStreamers.WebSockets.Handlers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Lykke.AlgoStore.Api.RealTimeStreaming.DataStreamers.WebSockets.Middleware
+{
+    public class CandlesWebSocketsMiddleware : WebSocketMiddlewareBase
```

Comment:
CandlesWebSocketsMiddleware, FunctionsWebSocketsMiddleware and TradesWebSocketsMiddleware have the exact same copy-pasted code, can you not use only one class?

Reply:
Please see the overall comment below.